File: leetcode/0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<>();

        for (int i = 0; i < numRows; i++) {
            List<Integer> list = new ArrayList<>();

            for (int j = 0; j <= i; j++) {
                // 양 끝은 항상 1
                if (j == 0 || j == i) {
                    list.add(1);
                } else {
                    int prev = result.get(i - 1).get(j - 1);
                    int next = result.get(i - 1).get(j);

                    list.add(prev + next);
                }
            }

            result.add(list);
        }

        return result;
    }
}

File: leetcode/0118-pascals-triangle/0118-pascals-triangle.ts
function generate(numRows: number): number[][] {
    const result: number[][] = [];

    for (let i = 0; i < numRows; i++) {
        const list: number[] = [];

        for (let j = 0; j <= i; j++) {
            // 양 끝은 항상 1
            if (j === 0 || j === i) {
                list.push(1);
            } else {
                const prev: number = result[i - 1][j - 1];
                const next: number = result[i - 1][j];

                list.push(prev + next);
            }
        }

        result.push(list);
    }

    return result;
}